core/primitives/src/types.rs

```diff
@@ -135,15 +157,19 @@ impl TransactionBody {
             TransactionBody::SendMoney(t) => t.nonce,
             TransactionBody::DeployContract(t) => t.nonce,
             TransactionBody::FunctionCall(t) => t.nonce,
+            TransactionBody::CreateAccount(t) => t.nonce,
+            TransactionBody::SwapKey(t) => t.nonce,
         }
     }
 
     pub fn get_sender(&self) -> AccountId {
         match self {
             TransactionBody::Stake(t) => t.staker,
             TransactionBody::SendMoney(t) => t.sender,
-            TransactionBody::DeployContract(t) => t.owner,
+            TransactionBody::DeployContract(t) => t.contract_id,
             TransactionBody::FunctionCall(t) => t.originator,
+            TransactionBody::CreateAccount(t) => t.sender,
```

Comment:
i'm in favor of originator if you are not actually sending something like in `SendMoney`

Reply:
It does send money though